File: crud-framework-core/src/main/java/com/antelopesystem/crudframework/components/componentmap/ComponentMap.java
package com.antelopesystem.crudframework.components.componentmap;


import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ComponentMap {

	Class<?> key() default void.class;

	Class<?> value() default void.class;
}

File: crud-framework-utils/src/main/java/com/antelopesystem/crudframework/utils/component/componentmap/annotation/ComponentMap.kt
package com.antelopesystem.crudframework.utils.component.componentmap.annotation

import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
annotation class ComponentMap(@Deprecated("no longer used") val key: KClass<*> = Unit::class, @Deprecated("no longer used") val value: KClass<*> = Unit::class)